packages/message-broker/src/index.spec.ts

```diff
@@ -0,0 +1,87 @@
+jest.mock('amqplib');
+jest.mock('@user-office-software/duo-logger');
+
+import { RabbitMQMessageBroker, Queue } from './index';
+import amqp from 'amqplib';
+
+describe('RabbitMQMessageBroker', () => {
+  const sleep = (ms = 0) => new Promise((resolve) => setTimeout(resolve, ms));
+  let broker: RabbitMQMessageBroker;
+  let mockAmqpChannel: jest.Mocked<Partial<amqp.Channel>>;
+  let mockAmqpConnection: jest.Mocked<Partial<amqp.Connection>>;
+  let mockAmqpConnectionEventCallbacks: Map<string, Function> = new Map();
+
+  beforeEach(() => {
+    mockAmqpChannel = {
+      assertQueue: jest.fn().mockResolvedValue({ queue: 'test-queue' }),
+      assertExchange: jest.fn(),
+      bindQueue: jest.fn(),
+      publish: jest.fn(),
+      consume: jest.fn().mockResolvedValue({}),
+      on: jest.fn(),
+    };
+
+    mockAmqpConnection = {
+      createChannel: jest.fn().mockResolvedValue(mockAmqpChannel),
+      close: jest.fn().mockResolvedValue(undefined),
+      on: jest.fn().mockImplementation((event, cb) => {
+        mockAmqpConnectionEventCallbacks.set(event, cb);
+      }),
+    };
+
+    amqp.connect = jest.fn().mockResolvedValue(mockAmqpConnection);
+
+    broker = new RabbitMQMessageBroker();
+  });
+
+  describe('registerConsumers', () => {
+    let consumer: jest.Mock;
+
+    beforeEach(async () => {
+      consumer = jest.fn();
+      await broker.setup();
+    });
+
+    it('should register a consumer when "listenOn" function is called', async () => {
+      broker.listenOn(Queue.SCHEDULING_PROPOSAL, consumer);
+
+      // waiting for registerConsumer function to finish behind the scenes
+      await sleep();
```

Comment:
I see you have declared` const sleep = (ms = 0) => new Promise((resolve) => setTimeout(resolve, ms));`
But I don't understand how does it work without passing any argument

Reply:
Ah, sorry, let me clarify my question..
What I wanted to ask was why do we put setTimeout with 0 seconds here, shouldn't it be 5 seconds or something?

Reply:
Yeah, it is due to the asynchronous task in the listenOn function. The listenOn function calls an asynchronous job, but it doesn't wait for it, and it also doesn't return with it, so we can't wait for the function to finish. I didn't want to change the interface because of backward compatibility; therefore, we need to ensure that the async job in the listenOn function finishes before the expect line. With a setTimeout zero delayed timer task, we can wait for the task and test the result.
Basically, it is about controlling the execution order of tasks in the Event Loop.

Reply:
Thanks for the good explanation!